src/pathspec.c

```diff
@@ -296,6 +296,8 @@ int git_pathspec_matches_path(
 
 static void pathspec_match_free(git_pathspec_match_list *m)
 {
+	assert(m);
```

Comment:
As this is a freeing function, we should handle the input being NULL by returning immediately, just like `free()` does.